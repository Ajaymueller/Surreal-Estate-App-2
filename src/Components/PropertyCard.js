import React from "react";
import PropTypes from "prop-types";
import '../styles/PropertyCard.css';
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  faBed,
  faBath,
  faPoundSign,
  faEnvelope,
  faHome,
} from "@fortawesome/free-solid-svg-icons";

const PropertyCard = ({
  _id,
  title,
  type,
  city,
  bathrooms,
  bedrooms,
  price,
  email,
  userID,
  onSaveProperty,
}) => {
  return (
    <div className="PropertyCard">
      <logo><FontAwesomeIcon icon={faHome} /></logo>
      <header>
        <h2 data-testid="title-id" className="title">
          {title}
          </h2>
        <h3 data-testid="type-city" className="type-city">
          {`${type} - ${city}`}
          </h3>
        {userID && (
          <button type="button" onClick={() => onSaveProperty(_id)}>
            Save
          </button>
        )}
      </header>
      <section>
        <div className="bathroom" data-testid="bathroom-id">
          <FontAwesomeIcon icon={faBath} /> 
          {`${bathrooms}`}</div>
        <div className="bedroom" data-testid="bedroom-id">
          <FontAwesomeIcon icon={faBed} /> 
          {`${bedrooms}`}</div>
        <div className="price" data-testid="price-id">
          <FontAwesomeIcon icon={faPoundSign} /> 
          {` ${price}`}</div>
      </section>
      <footer className="email" data-testid="email-id">
        <FontAwesomeIcon icon={faEnvelope} /> 
        <a href={`mailto:${email}`}>Email for details</a>
      </footer>
    </div>
  );
};

PropertyCard.propTypes = {
  _id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  city: PropTypes.string.isRequired,
  bathrooms: PropTypes.string.isRequired,
  bedrooms: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
  userID: PropTypes.string.isRequired,
  onSaveProperty: PropTypes.func.isRequired,
};

export default PropertyCard;