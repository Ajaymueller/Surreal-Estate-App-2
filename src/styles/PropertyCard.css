.PropertyCard {
    flex: 0 1 225px;
    display: flex;
    flex-direction: column;
    padding: 10px;
    gap: 5px;
    margin: 10px;
    font-family: "Open Sans"; 
    max-height: 500px;
    border: 5px solid white;
    box-sizing: border-box;
    font-size: 20px;
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
    text-align: center;
    background-color: #f1f1f1;
    color:#262626;
    border-radius: 10px;
    border: 5px solid white;
}

a:hover {
    color: green
}

logo {
    font-size: 75px;
}

h2 {
    font-weight: bold;
}

.bathroom {
    margin: 10px;
    padding: 5px;
    font-size: larger;
}

.bedroom {
    margin: 10px;
    padding: 5px;
    font-size: larger;
}

.price {
    margin: 10px;
    padding: 5px;
    font-size: larger;
}

.email {
    font-size: larger;
    margin: 10px;
    padding: 5px;
}