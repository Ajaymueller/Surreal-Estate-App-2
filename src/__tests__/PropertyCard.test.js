import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import PropertyCard from '../Components/PropertyCard';
import { MemoryRouter } from 'react-router-dom';
import "@testing-library/jest-dom/extend-expect";

const props = {
    title: "randomTitle", 
    type:"Semi-Detached", 
    bathrooms: 3, 
    bedrooms: 4, 
    price: 200000, 
    email: "random@email.com",
    _id: "id",
    userID: 12345, 
    onSaveProperty: jest.fn(),
};

describe("PropertyCard", () => {
    it("renders with correct props", () => {
     const { asFragment } = render(
    <MemoryRouter><PropertyCard {...props} /></MemoryRouter> )
    expect(asFragment).toMatchSnapshot();
    });
    it("has all correct fields", () => {
    const { getByTestId } = render(
    <MemoryRouter><PropertyCard {...props} /></MemoryRouter> )
    expect(getByTestId("title-id")).toHaveClass("title");
    expect(getByTestId("type-city")).toHaveClass("type-city");
    expect(getByTestId("bathroom-id")).toHaveClass("bathroom");
    expect(getByTestId("bedroom-id")).toHaveClass("bedroom");
    expect(getByTestId("price-id")).toHaveClass("price");
    expect(getByTestId("email-id")).toHaveClass("email");
    });
    it("displays a button", () => {
    const { getByRole } = render(
    <MemoryRouter><PropertyCard {...props} /></MemoryRouter> );
    const btn = getByRole("button");
    expect(btn).toBeInTheDocument();
    expect(btn).toHaveTextContent("Save");
    });
    it("calls onSaveProperty when button is clicked", () => {
    const { getByRole } = render(
    <MemoryRouter><PropertyCard {...props} /></MemoryRouter> )    
    const onSaveProperty = props.onSaveProperty;
    fireEvent.click(getByRole("button"));
    expect(onSaveProperty).toHaveBeenCalled();
    });
});