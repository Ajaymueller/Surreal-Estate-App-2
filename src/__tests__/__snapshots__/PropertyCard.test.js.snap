exports[`PropertyCard renders with correct props 1`] = `[Function]`;
